Automattic/woocommerce-payments
# Handle errors in retrieving a file gracefully

Fixes #4936

#### Changes proposed in this Pull Request

Trying to fetch a file may sometimes result in an exception, for example, when it doesn't exist or was deleted in the backend. For the merchant brand logo used on the printed receipt, not handling the exception leads to the rest of the receipt not being generated.

This PR adds error handling to ensure functionality dependent on the file API doesn't break because of failure to get a file.

#### Testing instructions

* Simulate a missing file error by prepending gibberish text to file id:

Change:
https://github.com/Automattic/woocommerce-payments/blob/a60e6343e4e6c9e6bbb8febb3f54cb6358987497/includes/wc-payment-api/class-wc-payments-api-client.php#L1021

To:
```php
return $this->request( [ 'as_account' => $as_account ], self::FILES_API . '/blablabla' . $file_id . '/contents', self::GET );
```

* Make a GET call to `http://localhost:8082/wp-json/wc/v3/payments/readers/receipts/<payment_intent_id>`.

* Verify that a 200 response is returned with HTML content for printed receipts (without a logo).

* **Before:** Receipt generation was interrupted by an error response.

-------------------


**Post merge**
